Add tests for Keras callbacks: MetricAverageCallback, LearningRateScheduleCallback and LearningRateWarmupCallback

14leeyuchieh opened the pull request:
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

There were no tests for MetricAverageCallback, LearningRateScheduleCallback and LearningRateWarmupCallback from
hvd as noted in #2659. This PR adds testing to verify the callback works.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

chongxiaoc commented on test/parallel/test_tensorflow2_keras.py:
```diff
@@ -16,27 +16,24 @@
+import numpy as np
 import pytest
-
+import tensorflow as tf
```
Is there a reason of moving the order of importing packages?
I think the convention is: 1. import third party packages; 2. from .. import third party packages; 3. horovod internal stuff.

14leeyuchieh replied:
It was auto-formatted via pylance on VSCode. I disabled it now in consideration of convention.

chongxiaoc commented on test/parallel/test_tensorflow2_keras.py:
```diff
@@ -173,6 +203,7 @@ class TestingOptimizer(optimizer_v2.OptimizerV2):
             """
             Custom optimizer we use for testing gradient aggregation.
             """
+
```
remove blank line?

chongxiaoc commented on test/parallel/test_tensorflow2_keras.py:
```diff
@@ -251,13 +282,17 @@ def test_process_set_optimizer(self):
         class TestOptimizer(keras.optimizers.Optimizer):
             def __init__(self, name, **kwargs):
                 super(TestOptimizer, self).__init__(name, **kwargs)
+
```
same as above.

chongxiaoc commented on test/parallel/test_tensorflow2_keras.py:
```diff
@@ -251,13 +282,17 @@ def test_process_set_optimizer(self):
             def get_gradients(self, loss, params):
                 assert len(params) == 1
                 return [tf.constant([float(hvd.rank())])]
+
```
same.

chongxiaoc reviewed (commented):
LGTM
Just a question about importing order and format.

Tixxx commented:
@14leeyuchieh thanks so much for adding the test. I think mostly it looks good, could you just add 1 more to test when multiplier is a function of epoch? It would be good to test that code path as well.

14leeyuchieh commented:
@Tixxx I added the test case now, and fixed the failure that was occurring. Can you approve the CI workflow again? Thank you.

```
[0]<stdout>:==================================== PASSES ====================================
[0]<stdout>:__________________ Tf2KerasTests.test_train_model_lr_schedule __________________
```

14leeyuchieh commented:
The test failure is unrelated. 
```
[0]<stdout>:test_torch.py::TorchTests::test_horovod_grouped_allreduce_process_sets FAILED
[0]<stderr>:terminate called after throwing an instance of 'gloo::IoException'
[0]<stderr>:  what():  [/tmp/pip-req-build-x8wd39an/horovod/common/gloo/http_store.cc:57] [/tmp/pip-req-build-x8wd39an/horovod/common/gloo/http_store.cc:57] Wait timeout after 30 seconds for key(s): 1. You may want to increase the timeout via HOROVOD_GLOO_TIMEOUT_SECONDS
[0]<stderr>:Fatal Python error: Aborted
[0]<stderr>:
```
